### PR TITLE
Improve masked attention

### PR DESCRIPTION
**Pull request by yozhikoff**
Hi!

I've just read the paper and it is excellent! Answered so many questions I've had recently.

I've been looking at the ABMIL implementation and I'm a bit confused by the masked attention part.
If `mask` is a binary mask, then the multiplication by it effectively sets the masked elements to zero. 
However, `softmax` will still treat zeros as full-fledged elements, "unmasking" them.

Setting masked elements to `-inf` solves the issue.

 ```python
>>> a = torch.tensor([0.43, -0.21, 0.99, 0.32, -0.35, 0.04])
>>> mask = torch.tensor([True, True, True, False, False, False])

>>> a_masked_multiplication = a * mask
>>> a_masked_multiplication
tensor([ 0.4300, -0.2100,  0.9900,  0.0000, -0.0000,  0.0000])
>>> torch.softmax(a_masked_multiplication, dim=0)
tensor([0.1912, 0.1008, 0.3348, 0.1244, 0.1244, 0.1244])

>>> a_masked_inf = torch.masked_fill(a, ~mask, -torch.inf)
>>> a_masked_inf
tensor([ 0.4300, -0.2100,  0.9900,    -inf,    -inf,    -inf])
>>> torch.softmax(a_masked_inf, dim=0)
tensor([0.3051, 0.1609, 0.5341, 0.0000, 0.0000, 0.0000])
```

Please close this PR if I'm misinterpreting the context!

### CODE REVIEW

**Comment by georg-wolflein** [comment]
Hi Artem, you are indeed correct -- well spotted! Thank you for proposing the fix, I will merge it right now.

Luckily this will not affect the results in the paper because I trained with a batch size of 1 so I didn't end up using the mask.

Anyways, thank you for your interest in the paper. I am planning on tidying up the code in the next few weeks, but if you have any more questions feel free to open another issue:)
